flutter/news/lib/dependencies_injection.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:convert';
+
+import 'package:get_it/get_it.dart';
+import 'package:news/features/frontpage/data/datasource/articles_local_data_source.dart';
+import 'package:news/features/frontpage/data/datasource/articles_remote_data_source.dart';
+import 'package:news/features/frontpage/data/repositories/articles_repository.dart';
+import 'package:news/features/frontpage/presentation/bloc/articles_cubit.dart';
+
+import 'core/http_client.dart';
+import 'core/news_api_client.dart';
+
+final getIt = GetIt.instance;
+
+Future<void> init() async {
+  getIt.registerFactory(() => ArticlesCubit(repository: getIt()));
+
+  getIt.registerLazySingleton(
+    () =>
+        ArticlesRepository(localDataSource: getIt(), remoteDataSource: getIt()),
+  );
+
+  getIt.registerLazySingleton(() => ArticlesLocalDataSource(getIt()));
```

Comment:
perhaps here the named arguments `MyStuff({required dependency, required dependency2})` could be useful to understand by reading this class what are the dependencies of things like the `ArticlesLocalDataSource` (etc).

Reply:
We should revisit this classes and make them have named arguments for the construction

Reply:
updated